apps/fabric-website/index.html

```diff
@@ -29,32 +28,107 @@
+      <img class="loadingImage" src="https://static2.sharepointonline.com/files/fabric/fabric-website/images/spinner.gif"
+        alt="Loading" width="32" height="32" />
     </div>
   </div>
 
   <script type="text/javascript">
-    function loadScript(n) {
-      for (var i = 0; i < n.length; i++) {
-        var s = document.createElement('script');
-        s.src = n[i];
-        s.async = false;
-        document.body.appendChild(s);
-      }
+    var isLocal = window.location.hostname === 'localhost' || window.location.hostname.indexOf('ngrok.io') > -1;
+    var isDogfood = false;
+    var isProduction = false;
+
+    // Query params
+    var dogfoodParam = getParameterByName('isDogfood');
+    var productionParam = getParameterByName('isProduction');
+    var noCompress = isLocal || getParameterByName('noCompress');
+    var devhost = getParameterByName('devhost');
+
+    var entryPointFilename = 'fabric-sitev5';
+    var appPath = '';
+    var Flight = {};
+
+    // Determine production or staging/dogfood
+    if (location.hostname === 'developer.microsoft.com' || productionParam) {
+      isProduction = true;
+    } else if (location.hostname === 'uifabric-tst.azurewebsites.net' || dogfoodParam) {
+      isDogfood = true;
+    }
+
+    // If the site is hosted, load the flight config script for production or dogfood
+    if (!isLocal) {
+      var configPrefix = isProduction ? 'fabric-website-prod' : 'fabric-website-df';
+      var configScript = ['https://fabricweb.azureedge.net/fabric-website/manifests/' + configPrefix + '.js'];
+
+      loadScripts(configScript, function () {
+        if (window.Flight) {
+          appPath += window.Flight.baseCDNUrl;
+
+          loadAppScripts(appPath);
+        }
+      });
+    }
+
+    else {
+      loadAppScripts(appPath);
+    }
+
+    // Simple synchronous script loader with callback. Adapted from https://stackoverflow.com/questions/1866717
+    function loadScripts(array, callback) {
+      var loader = function (src, handler) {
+        var script = document.createElement('script');
+        script.src = src;
+        script.onload = script.onreadystatechange = function () {
+          script.onreadystatechange = script.onload = null;
+          handler();
+        }
+        var head = document.getElementsByTagName('head')[0];
+        (head || document.body).appendChild(script);
+      };
+      (function run() {
+        if (array.length != 0) {
+          loader(array.shift(), run);
+        } else {
+          callback && callback();
+        }
+      })();
     }
 
-    var isLocal = window.location.hostname === 'localhost';
-    // This path allows minified files to be served locally for testing purposes.
-    var appPath = isLocal ? '' : '';
+    function loadAppScripts(appPath) {
+      var reactPath = '//cdnjs.cloudflare.com/ajax/libs/react/16.3.2/umd/react';
+      var reactDomPath = '//cdnjs.cloudflare.com/ajax/libs/react-dom/16.3.2/umd/react-dom';
```

Comment:
Just noting that we're bumping from `16.3.1 => 16.3.2` with this change

Reply:
Good callout. One thing to note here is that the public, hosted site currently doesn't use this method to load React--they're baked into the page by the server. I'd like to change that soon so we can completely self-manage this through Fabric. But in the meantime, it means that this piece will only be relevant for a.) local testing, and b.) the staging site I'm setting up to act as a dogfood environment.